### PR TITLE
motor updates

### PR DESCRIPTION
**Pull request by supertick**
Motor updates / clean up
Added Sabertooth motor controller config
Controllers are name references now 
Concrete linear mapper for serialization

### CODE REVIEW

File: src/main/java/org/myrobotlab/service/interfaces/AxisListener.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package org.myrobotlab.service.interfaces;
+
+import org.myrobotlab.framework.interfaces.NameProvider;
+import org.myrobotlab.service.data.AxisData;
+
+public interface AxisListener extends NameProvider {
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/service/interfaces/AxisListener.java]: this should have an attachAxisPublisher and AxisPublisher should be able to attachAxisListener

File: src/main/java/org/myrobotlab/service/interfaces/AxisPublisher.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package org.myrobotlab.service.interfaces;
+
+import org.myrobotlab.service.data.AxisData;
+
+public interface AxisPublisher {
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/service/interfaces/AxisPublisher.java]: this should have an attachAxisListener in this interface.

File: src/main/java/org/myrobotlab/service/interfaces/AxisListener.java (hunk 1 above)
Comment (supertick) [reply src/main/java/org/myrobotlab/service/interfaces/AxisListener.java]: fixed

File: src/main/java/org/myrobotlab/service/interfaces/AxisPublisher.java (hunk 2 above)
Comment (supertick) [reply src/main/java/org/myrobotlab/service/interfaces/AxisPublisher.java]: fixed

**Pull request merged**